_ml-commons-plugin/remote-models/batch-ingestion.md

```diff
@@ -0,0 +1,250 @@
+---
+layout: default
+title: Batch ingestion
+has_children: false
+nav_order: 71 
+parent: Connecting to externally hosted models 
+grand_parent: Integrating ML models
+redirect_from:
+  - /ml-commons-plugin/extensibility/batch-ingestion/
+---
+
+# Using externally hosted ML model for batch ingestion
+
+**Introduced 2.15**
+{: .label .label-purple }
+
+To ingest multiple documents which involve generating embeddings through external machine learning services, you can use OpenSearch's batch ingestion feature to achieve improved ingestion performance.
+
+The `_bulk` API accepts a `batch_size` parameter to indicate that documents in the bulk request should be grouped into batches with a size equal to the value of `batch_size`. Processors that support batch ingestion and connect to external machine learning services will consolidate documents from the same batch and send them in one request to the external service for inferencing.
+
+The `text_embedding` and `sparse_encoding` processors are two processors that support batch ingestion. When using these processors, setting the `batch_size` parameter in `_bulk` request allows them to take advantage of batched requests to the downstream machine learning services to improve ingestion throughput.
+
+## Step 1: Register a model group
+
+To register a model, you have the following options:
+
+* You can use `model_group_id` to register a model version to an existing model group.
+* If you do not use `model_group_id`, ML Commons creates a model with a new model group.
+
+To register a model group, send the following request:
+
+```
+POST /_plugins/_ml/model_groups/_register
+{
+  "name": "remote_model_group",
+  "description": "A model group for external models"
+}
+```
+{% include copy-curl.html %}
+
+The response contains the model group ID that you'll use to register a model to this model group:
+
+```
+{
+ "model_group_id": "wlcnb4kBJ1eYAeTMHlV6",
+ "status": "CREATED"
+}
+```
+
+To learn more about model groups, see [Model access control](https://github.com/opensearch-project/documentation-website/blob/7c4fe91ec9a16bb75e33726c2c86441edd56e08a/_ml-commons-plugin/remote-models/%7B%7Bsite.url%7D%7D%7B%7Bsite.baseurl%7D%7D/ml-commons-plugin/model-access-control).
+
+## Step 2: Create a connector
+
+You can create a standalone connector that can be reused for multiple models. Alternatively, you can specify a connector when creating a model so that it can be used only for that model. For more information and example connectors, see [Connectors](https://github.com/opensearch-project/documentation-website/blob/7c4fe91ec9a16bb75e33726c2c86441edd56e08a/_ml-commons-plugin/remote-models/%7B%7Bsite.url%7D%7D%7B%7Bsite.baseurl%7D%7D/ml-commons-plugin/remote-models/connectors).
```

Comment:
```suggestion
You can create a standalone connector that can be reused for multiple models. Alternatively, you can specify a connector when creating a model so that it can be used only for that model. For more information and example connectors, see [Connectors]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/connectors/).

```